apps/passport-server/src/services/generic-issuance/pipelines/PODPipeline/PODPipeline.ts

```diff
@@ -0,0 +1,439 @@
+        );
+
+        // @todo because this code is async, it may end up being interleaved
+        // with issuance requests at run-time. This may result in code
+        // requesting PCDs that - temporarily - do not exist because the atom
+        // DB has not been repopulated.
+        // A long-term solution might involve locking the pipeline or the DB,
+        // or refactoring so that this scenario cannot arise.
+        await this.db.clear(this.definition.id);
+        logs.push(makePLogInfo(`cleared old data`));
+        await this.db.save(this.definition.id, atoms);
+        logs.push(makePLogInfo(`saved parsed data: ${atoms.length} entries`));
+        span?.setAttribute("atom_count", atoms.length);
+
+        const end = new Date();
+
+        logs.push(
+          makePLogInfo(`load finished in ${end.getTime() - start.getTime()}ms`)
+        );
+
+        return {
+          lastRunStartTimestamp: start.toISOString(),
+          lastRunEndTimestamp: end.toISOString(),
+          latestLogs: logs,
+          atomsLoaded: atoms.length,
+          atomsExpected: atoms.length,
+          success: true
+        } satisfies PipelineLoadSummary;
+      } catch (e) {
+        setError(e, span);
+        logs.push(makePLogErr(`failed to load input: ${e}`));
+        logs.push(makePLogErr(`input was ${this.definition.options.input}`));
+
+        const end = new Date();
+        logs.push(
+          makePLogInfo(`load finished in ${end.getTime() - start.getTime()}ms`)
+        );
+
+        return {
+          atomsLoaded: 0,
+          atomsExpected: 0,
+          lastRunEndTimestamp: end.toISOString(),
+          lastRunStartTimestamp: start.toISOString(),
+          latestLogs: logs,
+          success: false
+        } satisfies PipelineLoadSummary;
+      }
+    });
+  }
+
+  public async start(): Promise<void> {
+    logger(`Starting POD Pipeline ${this.definition.id}`);
+  }
+
+  public async stop(): Promise<void> {
+    logger(`Stopping POD Pipeline ${this.definition.id}`);
+  }
+
+  /**
+   * Loads the atoms that match the given credential.
+   *
+   * @param credential The credential to match against
+   * @returns The atoms that match the given credential
+   */
+  private async loadMatchingAtoms({
+    email,
+    semaphoreId
+  }: VerifiedCredentialWithEmail): Promise<PODAtom[]> {
+    const atoms = await this.db.load(this.id);
+    const matchingAtoms: PODAtom[] = [];
+
+    for (const atom of atoms) {
+      if (atom.matchTo.type === "none") {
+        // No filter, so all atoms match to all credentials
+        matchingAtoms.push(atom);
+      } else if (atom.matchTo.type === "email") {
+        if (
+          atom.entries[atom.matchTo.entry].value.toString().toLowerCase() ===
+          email.toLowerCase()
+        ) {
+          matchingAtoms.push(atom);
+        }
+      } else if (atom.matchTo.type === "semaphoreID") {
+        if (
+          atom.entries[atom.matchTo.entry].value.toString() ===
```

Comment:
Why do these need to be compared as strings?  Is the raw type here bigint?

Reply:
The left side is `string | bigint` (either of the types that a `PODValue` can have) and the right side is `string`.

Reply:
Oh, okay, this is hiding a conversion error.  It's not particularly dangerous at this level, though note that a semaphore ID entry configured as a string won't match in proofs (which expect a cryptographic) so it might be best to force users to configure the correct type, or to make it so that the wrong type will not match (which I think is what the bare === will do).

Reply:
This is enforced in the UI, it just can't be inferred by TypeScript. I could use `as string` rather than `.toString()` though.

Reply:
That the entry used to match on a Semaphore ID is enforced by the UI, but this cannot be inferred by TypeScript. Since the source of the `semaphoreId` variable is ultimately `SemaphoreSignaturePCD`, which represents Semaphore IDs as strings, we need both to have a common type. I'll make this clearer with a comment.